### PR TITLE
Yandex speechkit api's supports custom IAmTokenProvider

### PR DESCRIPTION
**Pull request by kusraevs**

### CODE REVIEW

File: yandex-speechkit/src/main/java/com/justai/aimybox/speechkit/yandex/cloud/YandexRecognitionApi.kt [hunk 1]
```diff
@@ -17,7 +17,8 @@ internal class YandexRecognitionApi(
     private val yandexPassportOAuthKey: String,
```
Comment (morfeusys) [inline yandex-speechkit/src/main/java/com/justai/aimybox/speechkit/yandex/cloud/YandexRecognitionApi.kt]: I'd get rid of this argument and replace it with `iAmTokenProvider` without the default value - let the user to define it explicitly.

File: yandex-speechkit/src/main/java/com/justai/aimybox/speechkit/yandex/cloud/YandexSynthesisApi.kt [hunk 2]
```diff
@@ -12,7 +12,8 @@ import kotlin.coroutines.resumeWithException
 
 internal class YandexSynthesisApi(
     private val yandexPassportOAuthKey: String,
```
Comment (morfeusys) [inline yandex-speechkit/src/main/java/com/justai/aimybox/speechkit/yandex/cloud/YandexSynthesisApi.kt]: The same as for the previous comment

File: yandex-speechkit/src/main/java/com/justai/aimybox/speechkit/yandex/cloud/YandexRecognitionApi.kt (hunk 1 above)
Comment (kusraevs) [reply yandex-speechkit/src/main/java/com/justai/aimybox/speechkit/yandex/cloud/YandexRecognitionApi.kt]: Old YandexRecognitionApi contructor calls will not compile. Is it ok?

Comment (morfeusys) [reply yandex-speechkit/src/main/java/com/justai/aimybox/speechkit/yandex/cloud/YandexRecognitionApi.kt]: @kusraevs it's okay - we'll change the major version

**Pull request merged**